src/transformers/models/llava/processing_llava.py

```diff
@@ -134,3 +134,57 @@ def model_input_names(self):
         tokenizer_input_names = self.tokenizer.model_input_names
         image_processor_input_names = self.image_processor.model_input_names
         return list(dict.fromkeys(tokenizer_input_names + image_processor_input_names))
+
+    @property
+    def default_chat_template(self):
+        """
+        This default vicuna template formats inputs in the form of a chat history. For each message in the chat history:
+        * the template will output the role of the speaker followed by the content of the message.
+        * content is a list of strings and images.
+        * If the content element is an image, the template will output a sequence of <image> tokens
+
+        Example:
+
+        ```python
+        messages = [{
+            "role": "user",
+            "content": [
+                {"type": "text", "text": "What’s the content of this image?"},
+                {"type": "image"},
+                ],
+        },
+        {
+            "role": "assistant",
+            "content": [{"type": "text", "text": "This picture shows a red stop sign."},]
+        }]
+        ```
+
+        Will create outputs like:
+        ```
+        USER: <image>\nWhat is the content of this image? ASSITANT: This picture shows a red stop sign
```

Comment:
Why is the image token being place before the text, when the text comes first in the content list? If this is the case, then this should be detailed above in bullet points of behaviour

Reply:
Llava models need image to be placed before text, in all cases. I thought to make the template in a way that it format correctlly even if the users pass "image" in arbitrary order 

I can make it the same order as from user, then we need to update docs/readme on the hub explaining how to use the template correctly. WDYT, better to use user-order?